doc/specs/#653 - Quake Mode/#653 - Quake Mode.md

```diff
@@ -0,0 +1,661 @@
+but Teams is actually still running in the system tray, or the "notification
+area".
+
+![The Teams tray icon](tray-icon-000.png)
+
+_fig 1: an example of the Teams tray icon in the notification area_.
+
+When users want to be able to "minimize to the tray", they want:
+* The window to no longer appear on the taskbar
+* The window to no longer appear in the alt-tab order
+
+When minimized to the tray, it's almost as if there's no window for the Terminal
+at all. This can be combined with the global hotkey to quickly restore the window.
+
+The tray icon could be used for a variety of purposes. As a simple start, we
+could include the following three options:
+
+```
+Focus Terminal
+---
+Windows > Window 1 - <un-named window>
+          Window 2 - "This-window-does-have-a-name"
+---
+Quit
+```
+
+Just clicking on the icon would summon the recent terminal window. Right
+clicking would show the menu with "Focus Terminal", "Windows" and "Quit" in it, and
+"Windows" would have nested entries for each Terminal window.
+
+* "Focus Terminal" would do just that - summon the most recent terminal window,
+  wherever it is.
+* "Windows" would have nested popups for each open Terminal window. Each of
+  these nested entries would display the name and ID of the window. Clicking
+  them would summon that window (wherever it may be)
+* "Quit" would be akin to quit in browsers - close all open windows
+  <sup>[[1]](#footnote-1)</sup>.
+
+The tray notification would be visible always when the user has
+`"minimizeToTray": true` set in their settings. If the user has that set to
+false, but would still like the tray, they can specify `"alwaysShowTrayIcon":
+true`. That will cause the tray icon to always be added to the system tray.
+
+There's not a combination of settings where the Terminal is "minimized to the
+tray", and there's _no tray icon visible_. We don't want to let users get into a
+state where the Terminal is running, but is totally hidden from their control.
+
+From a technical standpoint, the tray icon is managed similar to the global
+hotkey. The Monarch process is responsible for setting it up, and processing the
+messages. When a Monarch dies and a new process becomes the Monarch, then it
+will re-create the tray icon.
+
+## UI/UX Design
+
+To summarize, we're proposing the following set of settings:
+
+```jsonc
+{
+    "minimizeToTray": bool,
+    "alwaysShowTrayIcon": bool,
+    "actions": [
+        {
+            "keys": KeyChord,
+            "command": {
+                "action": "globalSummon",
+                "dropdownDuration": float,
+                "hideWhenVisible": bool,
+                "monitor": "any"|"toCurrent"|"onCurrent"|int,
+                "desktop": "any"|"toCurrent"|"onCurrent"
+            }
+        },
+        {
+            "keys": KeyChord,
+            "command": {
+                "action": "quakeMode"
+            }
+        }
+    ]
+}
+```
+
+## Potential Issues

```

Comment:
Can there be a bit of discussion on WT Preview and WT Stable both having global hotkeys defined? Under the current design, the global hotkeys are assigned on settings reload, right? So if you reloaded WT Stable last, WT Preview's global hotkey will now be ignored.

Reply:
We're not exactly recommending that people run WTP and WT _at the same time_. I dunno.

Reply:
Yea our persistent guidance is "don't do that". If we wanted people to do that, then we'd have `wtp.exe` 😛

Reply:
Ok. We should make that clear in the spec/docs then.

The only other feature that has had this kind of problem in the past is execution aliases? And even then, I think the OS lets you configure it. So this one is definitely more dangerous.